File: src/services/tauri/official-site.ts
import { invoke } from '@tauri-apps/api/core'

export interface OfficialSiteBounds {
  x: number
  y: number
  width: number
  height: number
}

export function showOfficialSite(url: string, bounds: OfficialSiteBounds) {
  return invoke<string>('official_site_show', {
    url,
    ...bounds,
  })
}

export function navigateOfficialSite(url: string) {
  return invoke<string>('official_site_navigate', { url })
}

export function setOfficialSiteBounds(bounds: OfficialSiteBounds) {
  return invoke<void>('official_site_set_bounds', {
    ...bounds,
  })
}

export function hideOfficialSite() {
  return invoke<void>('official_site_hide')
}

export function reloadOfficialSite() {
  return invoke<void>('official_site_reload')
}

export function goBackOfficialSite() {
  return invoke<void>('official_site_go_back')
}

export function goForwardOfficialSite() {
  return invoke<void>('official_site_go_forward')
}

File: src-tauri/src/commands/official_site.rs
use crate::errors::AppError;
use crate::services::official_site;

#[tauri::command]
pub async fn official_site_show(
    app: tauri::AppHandle,
    url: String,
    x: f64,
    y: f64,
    width: f64,
    height: f64,
) -> Result<String, String> {
    official_site::show(&app, &url, x, y, width, height)
        .await
        .map_err(AppError::into_string)
}

#[tauri::command]
pub fn official_site_navigate(app: tauri::AppHandle, url: String) -> Result<String, String> {
    official_site::navigate(&app, &url).map_err(AppError::into_string)
}

#[tauri::command]
pub fn official_site_set_bounds(
    app: tauri::AppHandle,
    x: f64,
    y: f64,
    width: f64,
    height: f64,
) -> Result<(), String> {
    official_site::set_bounds(&app, x, y, width, height).map_err(AppError::into_string)
}

#[tauri::command]
pub fn official_site_hide(app: tauri::AppHandle) -> Result<(), String> {
    official_site::hide(&app).map_err(AppError::into_string)
}

#[tauri::command]
pub fn official_site_reload(app: tauri::AppHandle) -> Result<(), String> {
    official_site::reload(&app).map_err(AppError::into_string)
}

#[tauri::command]
pub fn official_site_go_back(app: tauri::AppHandle) -> Result<(), String> {
    official_site::go_back(&app).map_err(AppError::into_string)
}

#[tauri::command]
pub fn official_site_go_forward(app: tauri::AppHandle) -> Result<(), String> {
    official_site::go_forward(&app).map_err(AppError::into_string)
}
